feature/DOC-769: Document new resolve conversation options for the comments plugin (#1876)
tinymce/tinymce-docs: advanced/configuring-comments-callbacks.md

> **Note**: Failure to delete due to permissions or business rules is indicated by "false", while unexpected errors should be indicated using the "fail" callback.

### `tinycomments_resolve`

{{site.requires_5_8v}}

This option adds a _Resolve Conversation_ item to the dropdown menu of the first comment in a conversation.

> The `tinycomments_resolve` function should asynchronously return a flag indicating whether the comment or comment thread was resolved using the `done` callback. Unrecoverable errors are communicated to {{site.productname}} by calling the `fail` callback instead.

Reviewer: Is this correct? Resolve only applies to the comment thread IIUC.

> **Note**: Failure to delete due to permissions or business rules is indicated by "false", while unexpected errors should be indicated using the "fail" callback.

### `tinycomments_resolve`

> 

Reviewer 2: ```suggestion

{{site.requires_5_8v}}

```

By adding this option, a _Resolve Conversation_ item will be added to the dropdown menu of the first comment in a conversation.

The `tinycomments_resolve` function should asynchronously return a flag indicating whether the comment/comment thread was resolved using the `done` callback. Unrecoverable errors are communicated to {{site.productname}} by calling the `fail` callback instead.

> The `tinycomments_resolve` function is given a request (req) object as the first parameter, which has this field:

Reviewer 2: ```suggestion
The `tinycomments_resolve` function is passed a (`req`) object as the first parameter, which contains the following key-value pair:
```

> * **conversationUid**: The uid of the conversation the reply is a part of.

Reviewer 2: ```suggestion
* **conversationUid**: The uid of the conversation the reply is targeting.
```

Reviewer 2: If you can think of a good alternative for 'targeting', use it instead please.

> The done callback needs to take an object of the form:

Reviewer 2: ```suggestion
The `done` callback should accept the following object:
```

```js
{
  canResolve: boolean // whether or not the conversation can be resolved
  reason: string? // an optional string explaining why resolving was not allowed (if canResolve is false)
}
```

> > **Note**: Failure to resolve due to permissions or business rules is indicated by "false", while unexpected errors should be indicated using the "fail" callback.

Reviewer 2: ```suggestion
> **Note**: Failure to resolve due to permissions or business rules should be indicated by `canResolve: false`, while unexpected errors should be indicated using the `fail` callback.
```
